src/app/modules/persistence-service/persistence.service.ts

```diff
@@ -19,59 +24,226 @@ export abstract class AbstractPersistenceService {
   abstract getBinaryResource(path: string): Promise<Blob>;
 }
 
+export interface BackupEntry {
+  resource: string;
+  backupResource: string;
+}
+
+export interface PullResponse {
+  failure: boolean;
+  diffExists: boolean;
+  headCommit: string;
+  changedResources: Array<string>;
+  backedUpResources: Array<BackupEntry>;
+}
+
 @Injectable()
 export class PersistenceService extends AbstractPersistenceService {
 
   private serviceUrl: string;
   private listFilesUrl: string;
+  private openNonDirtyTabs: Array<string>;
```

Comment:
Would it be an option to use `Set`s instead? I would imagine it could simplify the code a bit, since you wouldn't have to manually check for duplicates…

Reply:
much better